satpy/readers/viirs_l2.py

```diff
@@ -0,0 +1,155 @@
+        return self.get(ds_id['name'] + '/shape', 1)
+
+    @property
+    def start_time(self):
+        """Get start time."""
+        return self._parse_datetime(self['/attr/time_coverage_start'])
+
+    @property
+    def end_time(self):
+        """Get end time."""
+        return self._parse_datetime(self['/attr/time_coverage_end'])
+
+    def get_metadata(self, dataset_id, ds_info):
+        """Get metadata."""
+        var_path = ds_info['file_key']
+        shape = self.get_shape(dataset_id, ds_info)
+        file_units = self._get_dataset_file_units(ds_info, var_path)
+
+        i = getattr(self[var_path], 'attrs', {})
+        i.update(ds_info)
+        i.update(dataset_id.to_dict())
+        i.update({
+            "shape": shape,
+            "units": ds_info.get("units", file_units),
+            "file_units": file_units,
+            "platform_name": self.platform_name,
+            "sensor": self.sensor_name,
+            "start_orbit": self.start_orbit_number,
+            "end_orbit": self.end_orbit_number,
+        })
+        i.update(dataset_id.to_dict())
+        return i
+
+    def _get_dataset_file_units(self, ds_info, var_path):
+        file_units = ds_info.get('file_units')
+        if file_units is None:
+            file_units = self.get(var_path + '/attr/units')
+
+        return file_units
+
+    def get_dataset(self, dataset_id, ds_info):
+        """Get dataset."""
+        var_path = ds_info['file_key']
+        metadata = self.get_metadata(dataset_id, ds_info)
+
+        valid_min, valid_max = self._get_dataset_valid_range(var_path)
+        data = self[var_path]
+        data.attrs.update(metadata)
+
+        if valid_min is not None and valid_max is not None:
+            data = data.where((data >= valid_min) & (data <= valid_max))
+
+        if isinstance(data.attrs.get('flag_meanings'), str):
+            data.attrs['flag_meanings'] = data.attrs['flag_meanings'].split(' ')
+
+        # rename dimensions to correspond to satpy's 'y' and 'x' standard
+        if 'Rows' in data.dims:
+            data = data.rename({'Rows': 'y', 'Columns': 'x'})
+        return data
+
+    def _get_dataset_valid_range(self, var_path):
+        valid_range = self.get(var_path + '/attr/valid_range')
+        valid_min = valid_range[0]
+        valid_max = valid_range[1]
+
+        return valid_min, valid_max
+
+    def available_datasets(self, configured_datasets=None):
+        """Generate dataset info and their availablity.
+
+        See
+        :meth:`satpy.readers.file_handlers.BaseFileHandler.available_datasets`
+        for details.
+
+        """
+        for is_avail, ds_info in (configured_datasets or []):
+            if is_avail is not None:
+                # some other file handler said it has this dataset
+                # we don't know any more information than the previous
+                # file handler so let's yield early
+                yield is_avail, ds_info
+                continue
```

Comment:
I don't quite understand what's going on here, but it seems this could also use a test.

Reply:
Neither do I. This was in the file handler I used as a template.

Reply:
If you define all the datasets in the yaml file, this function shouldn't be needed at all.